core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -766,6 +777,52 @@ void checkTaskTimeout(TaskDef taskDef, Task task) {
         timeoutTaskWithTimeoutPolicy(reason, taskDef, task);
     }
 
+    @VisibleForTesting
+    void checkTaskPublishTimeout(TaskDef taskDef, Task task) {
+        if (!task.getStatus().equals(SCHEDULED) || task.getScheduledTime() == 0) {
+            return;
+        }
+
+        if (!task.getWorkflowTask().getType().equals(TaskType.SIMPLE.name())) {
+            return;
+        }
+
+        long currentTime = System.currentTimeMillis();
+        long publishDuration = currentTime - task.getScheduledTime();
+        if (task.getLastPublishTime() > 0) {
+            publishDuration = currentTime - task.getLastPublishTime();
+        }
+        if (publishDuration > taskPublishTimeout) {
+            if (task.getPublishCount() >= MAX_PUBLISH_COUNT) {
+                long terminationTime = 181 * 24 * 3600 * 1000L; // 181 days
+                if (publishDuration > terminationTime) {
```

Comment:
after exceeding 3 attempts, task is moved to TIME_OUT state (which is fine). But workflow is terminated after 181 days is it ? We should change only Task status or is it fine to change workflow status to terminated ?

Reply:
as Manju said, its a system error condition that that status is not marked as IN_PROGRESS from SCHEDULED.